src/mongo/base/DriverBase.class.php

```diff
@@ -376,6 +378,38 @@ protected function getCollection()
         return $this->collection;
     }
 
+    protected function applyPreHooks($hooks,$args=array())
+    {
+        $this->applyHooks("pre",$hooks,$args);
+    }
+
+    protected function applyPostHooks($hooks,$args=array())
+    {
+        $this->applyHooks("post",$hooks,$args);
+    }
+
+    protected function applyFailureHooks($hooks,$args=array())
+    {
+        $this->applyHooks("failure",$hooks,$args);
+    }
```

Comment:
It would be good to have IEventHook::TIMING_PRE, TIMING_POST, TIMING_FAILURE, (I'm open as to the naming), so that we can comfortably call applyHooks directly, and do away with these indirect methods.